src/erl.rs

```diff
@@ -1,8 +1,14 @@
+mod format;
+mod literals;
+pub mod names;
 #[cfg(test)]
 mod tests;
+mod typespec;
 
 use crate::{
     ast::*,
+    erl::format::wrap_args,
+    erl::literals::atom,
```

Comment:
If possible I'd prefer to have refactoring the erl module into multiple modules to be a different PR. This one is gunna be pretty large just with the feature so to make it easier to review and merge it'd be good to try to avoid including other changes.

Reply:
I completely agree. I didn't want to be insubordinate :) I assumed that Rust didn't support cyclic imports with `erl` importing `erl::typespec` and `erl::typespec` importing `erl`. I didn't check though. If Rust supports them then I'll role them back into `erl` happily enough.

Reply:
Oh I see! I think it probably doesn't allow that? Maybe it does! Let's find out

Reply:
It seems that Rust doesn't mind having the cross dependencies which weirds me out a bit but it means we can leave everything where it was.